examples/bash_automatic_cd.sh

```diff
@@ -7,14 +7,13 @@
 # To undo the effect of this function, you can type "cd -" to return to the
 # original directory.
 
-function ranger-cd {
-    tempfile="$(mktemp -t tmp.XXXXXX)"
-    ranger --choosedir="$tempfile" "${@:-$(pwd)}"
-    test -f "$tempfile" &&
-    if [ "$(cat -- "$tempfile")" != "$(echo -n `pwd`)" ]; then
-        cd -- "$(cat "$tempfile")"
+ranger_cd() {
+    temp_file="$(mktemp -t "ranger_cd.XXXXXXXXXX")"
+    ranger --choosedir="$temp_file" -- "${@:-$PWD}"
```

Comment:
> we don't need any nesting which `$(pwd)` eases

Could you please explain this in more detail?

> I seem to recall some cases where `$PWD` isn't defined.

For example? `$PWD` is defined by POSIX.

Reply:
> Could you please explain this in more detail?

I meant that if we'd need something like `$(cat $(echo somefile))` (I know, a silly example), `$()` is much easier to nest and so often preferred. Since we only need a single level of command substitution, let's go with the older syntax.

Sorry for making it seem more complex than it is. :)

> For example? $PWD is defined by POSIX.

Sadly, I don't remember the details but I remember explicitly preferring `` `pwd` `` in some code I wrote some time ago.

On a second thought, considering I really doubt it all would run on a minimal shell like the original sh and would need bash/zsh instead, it should be fine.

Reply:
I guess it'll be ok too and it'll hardly be a loss of any compat considering we were already using this syntax before.

Reply:
To clarify: I mean that I'm fine with `$()` too.